lib/ClangImporter/ImportDecl.cpp

```diff
@@ -345,6 +345,23 @@ static bool isNSDictionaryMethod(const clang::ObjCMethodDecl *MD,
   return true;
 }
 
+static void forEachDistinctName(
+    ClangImporter::Implementation &impl, const clang::NamedDecl *decl,
+    llvm::function_ref<void(ImportedName, ImportNameVersion)> action) {
+  using ImportNameKey = std::pair<DeclName, EffectiveClangContext>;
+  SmallVector<ImportNameKey, 8> seenNames;
```

Comment:
EffectiveClangContext isn't orderable, so I can't use a normal set. But with N < 8 ImportNameVersions and most of them being the same I'm not too worried about N^2 behavior here.

I think I'm going to not worry about UnresolvedNames for now.